host/frontend/host-orchestrator/operation_test.go

```diff
@@ -117,48 +116,97 @@ func TestMapOMCompleteOperation(t *testing.T) {
 
 	t.Run("does not exist", func(t *testing.T) {
 		om := NewMapOM(uuidFactory)
-		result := OperationResultData{
-			Error: OperationErrorData{"error"},
+		result := OperationResult{
+			Error: OperationResultError{"error"},
+		}
+
+		om.Complete(opName, result)
+
+		_, err := om.Get(opName)
+		if err == nil {
+			t.Errorf("expected error")
 		}
+	})
+}
+
+func TestMapOMWaitOperation(t *testing.T) {
+	opName := "operation-1"
+	uuidFactory := func() string { return opName }
+	result := OperationResult{
+		Error: OperationResultError{"error"},
+	}
 
-		om.CompleteOperation(opName, result)
+	t.Run("stops waiting", func(t *testing.T) {
+		om := NewMapOM(uuidFactory)
+		om.New()
+		waitDoneCh := make(chan struct{}, 1)
+
+		go func() {
+			om.Wait(opName)
+			om.Wait(opName)
+			waitDoneCh <- struct{}{}
+		}()
+
+		om.Complete(opName, result)
+
+		select {
+		case <-waitDoneCh:
+			return
+		case <-time.After(1 * time.Second):
+			t.Errorf("expected stop waiting as operatioin was completed")
+			return
+		}
```

Comment:
most of this code could be encapsulated in a function to avoid repetition with the other test

Reply:
acknowledged, done